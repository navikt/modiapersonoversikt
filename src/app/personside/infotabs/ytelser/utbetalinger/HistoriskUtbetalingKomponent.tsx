import { HistoriskUtbetaling } from '../../../../../models/ytelse/ytelse-utbetalinger';
import { formaterDato } from '../../../../../utils/dateUtils';
import { formaterNOK } from '../../utbetalinger/utils/utbetalingerUtils';
import { getFormatertKreditortrekkFraHistoriskUtbetaling } from './utbetalingerUtils';
import * as React from 'react';
import { useState } from 'react';
import DetaljerCollapse from '../../../../../components/DetaljerCollapse';
import DescriptionList from '../../../../../components/DescriptionList';

interface Props {
    historiskUtbetaling: HistoriskUtbetaling;
}

function HistoriskUtbetalingKomponent({ historiskUtbetaling }: Props) {
    const utbetaltEntries = {
        Type: historiskUtbetaling.type,
        Netto: historiskUtbetaling.nettobeløp && formaterNOK(historiskUtbetaling.nettobeløp) + ' NOK',
        Dato: historiskUtbetaling.utbetalingsdato && formaterDato(historiskUtbetaling.utbetalingsdato),
        Periode:
            historiskUtbetaling.vedtak &&
            `${formaterDato(historiskUtbetaling.vedtak.fra)} - ${formaterDato(historiskUtbetaling.vedtak.til)}`,
        Utbetalingsgrad: historiskUtbetaling.utbetalingsgrad && historiskUtbetaling.utbetalingsgrad + '%'
    };
    const utbetaltDetaljerEntries = {
        Dagsats: historiskUtbetaling.dagsats && formaterNOK(historiskUtbetaling.dagsats),
        Bruttobeløp: historiskUtbetaling.bruttobeløp && formaterNOK(historiskUtbetaling.bruttobeløp) + ' NOK',
        Arbeidsgiver: historiskUtbetaling.arbeidsgiverNavn,
        Organisasjonsnummer: historiskUtbetaling.arbeidsgiverOrgNr,
        Skattetrekk: historiskUtbetaling.skattetrekk,
        Kreditortrekk: getFormatertKreditortrekkFraHistoriskUtbetaling(historiskUtbetaling)
    };
    const [visDetaljer, setVisDetaljer] = useState(false);
    return (
        <li aria-label={'Utført utbetaling ' + utbetaltEntries.Dato}>
            <DetaljerCollapse
                header={<DescriptionList entries={utbetaltEntries} />}
                alwaysGrayBackground={true}
                open={visDetaljer}
                toggle={() => setVisDetaljer(!visDetaljer)}
            >
                <DescriptionList entries={utbetaltDetaljerEntries} />
            </DetaljerCollapse>
        </li>
    );
}

export default HistoriskUtbetalingKomponent;
