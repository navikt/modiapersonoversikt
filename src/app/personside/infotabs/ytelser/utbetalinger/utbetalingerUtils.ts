import { HistoriskUtbetaling, KreditorTrekk } from '../../../../../models/ytelse/ytelse-utbetalinger';
import {
    flatMapYtelser,
    formaterNOK,
    getBruttoSumYtelser,
    getNettoSumYtelser,
    getTypeFromYtelse,
    reduceUtbetlingerTilYtelser
} from '../../utbetalinger/utils/utbetalingerUtils';
import { Utbetaling, Ytelse } from '../../../../../models/utbetalinger';
import { YtelserKeys } from '../ytelserKeys';

export function getFormatertKreditortrekkFraHistoriskUtbetaling(historiskUtbetaling: HistoriskUtbetaling): string {
    if (!historiskUtbetaling.trekk || historiskUtbetaling.trekk.length === 0) {
        return '';
    }
    const trekkSummert = historiskUtbetaling.trekk.reduce((acc: number, trekk: KreditorTrekk) => acc + trekk.beløp, 0);
    return formaterNOK(trekkSummert) + ' NOK';
}

export function mapUtbetalingerTilHistoriskeUtbetalinger(utbetalinger: Utbetaling[]): HistoriskUtbetaling[] {
    return utbetalinger.reduce(
        (acc: HistoriskUtbetaling[], utbetaling: Utbetaling) => [
            ...acc,
            ...mapUtbetalingTilHistoriskeUtbetalinger(utbetaling)
        ],
        []
    );
}

function mapUtbetalingTilHistoriskeUtbetalinger(utbetaling: Utbetaling): HistoriskUtbetaling[] {
    const alleTilhørendeYtelser = flatMapYtelser([utbetaling]);
    return alleTilhørendeYtelser.map(ytelse => mapYtelseTilHistoriskUtbetaling(ytelse, utbetaling));
}

function mapYtelseTilHistoriskUtbetaling(ytelse: Ytelse, utbetaling: Utbetaling): HistoriskUtbetaling {
    const nettoSum = getNettoSumYtelser([ytelse]);
    const bruttoSum = getBruttoSumYtelser([ytelse]);
    return {
        vedtak: {
            fra: ytelse.periode.start,
            til: ytelse.periode.slutt
        },
        utbetalingsgrad: null,
        utbetalingsdato: utbetaling.utbetalingsdato,
        nettobeløp: nettoSum,
        bruttobeløp: bruttoSum,
        skattetrekk: ytelse.skattsum,
        arbeidsgiverNavn: null,
        arbeidsgiverOrgNr: null,
        dagsats: null,
        type: ytelse.type,
        trekk: mapTrekkTilKreditortrekk(ytelse)
    };
}

function mapTrekkTilKreditortrekk(ytelse: Ytelse): null | KreditorTrekk[] {
    if (!ytelse.trekkListe) {
        return null;
    }
    return ytelse.trekkListe.map(trekk => {
        return {
            kreditorsNavn: trekk.kreditor || '',
            beløp: trekk.trekkbeløp
        };
    });
}

export function fjernUrelevanteUtbetalinger(relevantYtelse: YtelserKeys) {
    return (utbetaling: Utbetaling) => {
        const ytelser = reduceUtbetlingerTilYtelser([utbetaling]).filter(
            ytelse => relevantYtelse === getTypeFromYtelse(ytelse)
        );
        return {
            ...utbetaling,
            ytelser: ytelser
        };
    };
}
