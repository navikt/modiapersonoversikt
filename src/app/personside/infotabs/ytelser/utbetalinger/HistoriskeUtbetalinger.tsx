import { HistoriskUtbetaling } from '../../../../../models/ytelse/ytelse-utbetalinger';
import * as React from 'react';
import HistoriskUtbetalingKomponent from './HistoriskUtbetalingKomponent';
import { Knapp } from 'nav-frontend-knapper';
import NavFrontendSpinner from 'nav-frontend-spinner';
import {
    filtrerBortUtbetalingerSomIkkeErUtbetalt,
    fjernTommeUtbetalinger
} from '../../utbetalinger/utils/utbetalingerUtils';
import { genericAscendingDateComparator } from '../../../../../utils/dateUtils';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { YtelserKeys } from '../ytelserKeys';
import { AlignTextCenter, FlexCenter } from '../../../../../components/common-styled-components';
import { fjernUrelevanteUtbetalinger, mapUtbetlaingerTilHistoriskeUtbetalinger } from './utbetalingerUtils';
import { STATUS } from '../../../../../redux/restReducers/utils';
import ErrorBoundary from '../../../../../components/ErrorBoundary';
import { Utbetaling } from '../../../../../models/utbetalinger';
import { Undertittel } from 'nav-frontend-typografi';
import styled from 'styled-components';

interface Props {
    historiskeUtbetalinger: HistoriskUtbetaling[];
    ytelseType: YtelserKeys;
    toÅrGamleUtbetalingerFraUtbetalingerRestkonto: Utbetaling[] | null;
    reducerStatus: STATUS;
    hentHistoriskeUtbetalinger: () => void;
}

const Padding = styled.div`
    padding: 0.5rem;
`;

class HistoriskeUtbetalinger extends React.PureComponent<Props> {
    constructor(props: Props) {
        super(props);
    }

    getSammenflettedeUtbetalinger(): HistoriskUtbetaling[] {
        if (!this.props.toÅrGamleUtbetalingerFraUtbetalingerRestkonto) {
            return this.props.historiskeUtbetalinger;
        } else {
            const relevanteUtbetlaingerFraUtbetalingRestkonto = this.props.toÅrGamleUtbetalingerFraUtbetalingerRestkonto
                .map(fjernUrelevanteUtbetalinger(this.props.ytelseType))
                .filter(filtrerBortUtbetalingerSomIkkeErUtbetalt)
                .filter(fjernTommeUtbetalinger);

            const mappetTilHistoriskUtbetaling = mapUtbetlaingerTilHistoriskeUtbetalinger(
                relevanteUtbetlaingerFraUtbetalingRestkonto
            );

            return [...this.props.historiskeUtbetalinger, ...mappetTilHistoriskUtbetaling];
        }
    }

    render() {
        const sorterteSammenflettedeUtbetalinger = this.getSammenflettedeUtbetalinger()
            .sort(genericAscendingDateComparator(utbetaling => utbetaling.utbetalingsdato || new Date()))
            .reverse();

        if (this.props.reducerStatus === STATUS.SUCCESS && sorterteSammenflettedeUtbetalinger.length === 0) {
            return <AlertStripeInfo>Kunne ikke finne noen historiske utbetalinger</AlertStripeInfo>;
        }

        const historiskeUtbetalinger = sorterteSammenflettedeUtbetalinger.map((utbetaling, index) => (
            <HistoriskUtbetalingKomponent key={index} historiskUtbetaling={utbetaling} />
        ));

        const hentAlleUtbetalingerKnapp = this.props.reducerStatus === STATUS.NOT_STARTED && (
            <FlexCenter>
                <Knapp onClick={this.props.hentHistoriskeUtbetalinger}>Hent alle historiske utbetalinger</Knapp>
            </FlexCenter>
        );

        const spinner = this.props.reducerStatus === STATUS.LOADING && <NavFrontendSpinner type={'S'} />;

        return (
            <ErrorBoundary boundaryName="Historiske utbetalinger">
                <section>
                    <Padding>
                        <AlignTextCenter>
                            <Undertittel tag="h4">Utførte utbetalinger</Undertittel>
                        </AlignTextCenter>
                    </Padding>
                    <ol>{historiskeUtbetalinger}</ol>
                    {hentAlleUtbetalingerKnapp}
                    {spinner}
                </section>
            </ErrorBoundary>
        );
    }
}

export default HistoriskeUtbetalinger;
