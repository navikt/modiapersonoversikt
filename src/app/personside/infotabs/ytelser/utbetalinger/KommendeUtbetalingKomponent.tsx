import { KommendeUtbetaling } from '../../../../../models/ytelse/ytelse-utbetalinger';
import { formaterDato } from '../../../../../utils/dateUtils';
import { formaterNOK } from '../../utbetalinger/utils/utbetalingerUtils';
import * as React from 'react';
import { useState } from 'react';
import DescriptionList from '../../../../../components/DescriptionList';
import DetaljerCollapse from '../../../../../components/DetaljerCollapse';

interface Props {
    kommendeUtbetaling: KommendeUtbetaling;
}

function KommendeUtbetalingKomponent({ kommendeUtbetaling }: Props) {
    const kommendeEntries = {
        Registeringsdato: kommendeUtbetaling.utbetalingsdato && formaterDato(kommendeUtbetaling.utbetalingsdato),
        Type: kommendeUtbetaling.type,
        Periode:
            kommendeUtbetaling.vedtak &&
            `${formaterDato(kommendeUtbetaling.vedtak.fra)} - ${formaterDato(kommendeUtbetaling.vedtak.til)}`,
        Utbetalingsgrad: kommendeUtbetaling.utbetalingsgrad && kommendeUtbetaling.utbetalingsgrad + '%'
    };

    const kommendeDetaljerEntries = {
        Dagsats: kommendeUtbetaling.dagsats && formaterNOK(kommendeUtbetaling.dagsats) + ' NOK',
        Bruttobeløp: kommendeUtbetaling.bruttobeløp && formaterNOK(kommendeUtbetaling.bruttobeløp) + ' NOK',
        Arbeidsgiver: kommendeUtbetaling.arbeidsgiverNavn,
        Organisasjonsnummer: kommendeUtbetaling.arbeidsgiverOrgNr,
        'Saksbehandlerident (Tryde-ident)': kommendeUtbetaling.saksbehandler
    };
    const [visDetaljer, setVisDetaljer] = useState(false);
    return (
        <li aria-label={'Kommende utbetaling ' + kommendeEntries.Registeringsdato}>
            <DetaljerCollapse
                header={<DescriptionList entries={kommendeEntries} />}
                alwaysGrayBackground={true}
                open={visDetaljer}
                toggle={() => setVisDetaljer(!visDetaljer)}
            >
                <DescriptionList entries={kommendeDetaljerEntries} />
            </DetaljerCollapse>
        </li>
    );
}

export default KommendeUtbetalingKomponent;
