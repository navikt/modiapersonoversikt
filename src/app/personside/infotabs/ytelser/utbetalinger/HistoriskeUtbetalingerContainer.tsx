import * as React from 'react';
import { AppState } from '../../../../../redux/reducers';
import { AsyncDispatch } from '../../../../../redux/ThunkTypes';
import { hentHistoriskeUtbetalinger } from '../../../../../redux/restReducers/ytelser/historiskeUtbetalinger';
import { connect } from 'react-redux';
import { isLoaded, isNotStarted, Loaded, RestReducer } from '../../../../../redux/restReducers/restReducer';
import { UtbetalingerResponse } from '../../../../../models/utbetalinger';
import { Person, PersonRespons } from '../../../../../models/person/person';
import { YtelserKeys } from '../ytelserKeys';
import HistoriskeUtbetalinger from './HistoriskeUtbetalinger';
import { HistoriskUtbetaling } from '../../../../../models/ytelse/ytelse-utbetalinger';

interface OwnProps {
    ytelseType: YtelserKeys;
    historiskeUtbetalinger: HistoriskUtbetaling[];
}

interface StateProps {
    toÅrGamleUtbetalingerFraUtbetalingerRestkonto: RestReducer<UtbetalingerResponse>;
    personInformasjon: RestReducer<PersonRespons>;
}

interface DispatchProps {
    hentHistoriskeUtbetalinger: (fnr: string) => void;
}

type Props = DispatchProps & StateProps & OwnProps;

class HistoriskeUtbetalingerContainer extends React.PureComponent<Props> {
    constructor(props: Props) {
        super(props);
        this.hentAlleHistoriskeUtbetalinger = this.hentAlleHistoriskeUtbetalinger.bind(this);
    }

    hentAlleHistoriskeUtbetalinger() {
        if (
            isNotStarted(this.props.toÅrGamleUtbetalingerFraUtbetalingerRestkonto) &&
            isLoaded(this.props.personInformasjon)
        ) {
            const fnr = (this.props.personInformasjon as Loaded<Person>).data.fødselsnummer;
            this.props.hentHistoriskeUtbetalinger(fnr);
        }
    }

    render() {
        const toÅrGamleUtbetalingerFraUtbetalingerRestkonto = isLoaded(
            this.props.toÅrGamleUtbetalingerFraUtbetalingerRestkonto
        )
            ? this.props.toÅrGamleUtbetalingerFraUtbetalingerRestkonto.data.utbetalinger
            : null;
        return (
            <HistoriskeUtbetalinger
                historiskeUtbetalinger={this.props.historiskeUtbetalinger}
                ytelseType={this.props.ytelseType}
                hentHistoriskeUtbetalinger={this.hentAlleHistoriskeUtbetalinger}
                reducerStatus={this.props.toÅrGamleUtbetalingerFraUtbetalingerRestkonto.status}
                toÅrGamleUtbetalingerFraUtbetalingerRestkonto={toÅrGamleUtbetalingerFraUtbetalingerRestkonto}
            />
        );
    }
}

function mapStateToProops(state: AppState): StateProps {
    return {
        toÅrGamleUtbetalingerFraUtbetalingerRestkonto: state.restEndepunkter.historiskeUtbetalingerYtelser,
        personInformasjon: state.restEndepunkter.personinformasjon
    };
}

function mapDispatchToProps(dispatch: AsyncDispatch): DispatchProps {
    return {
        hentHistoriskeUtbetalinger: (fnr: string) => dispatch(hentHistoriskeUtbetalinger(fnr))
    };
}

export default connect(
    mapStateToProops,
    mapDispatchToProps
)(HistoriskeUtbetalingerContainer);
