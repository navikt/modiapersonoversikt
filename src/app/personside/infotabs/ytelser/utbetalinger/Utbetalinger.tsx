import * as React from 'react';
import { HistoriskUtbetaling, KommendeUtbetaling } from '../../../../../models/ytelse/ytelse-utbetalinger';
import styled from 'styled-components';
import theme from '../../../../../styles/personOversiktTheme';
import KommendeUtbetalinger from './KommendeUtbetalinger';
import { YtelserKeys } from '../ytelserKeys';
import HistoriskeUtbetalingerContainer from './HistoriskeUtbetalingerContainer';
import ErrorBoundary from '../../../../../components/ErrorBoundary';

interface Props {
    kommendeUtbetalinger: KommendeUtbetaling[];
    historiskeUtbetalinger: HistoriskUtbetaling[];
    ytelsesType: YtelserKeys;
}

const StyledSection = styled.section`
    > * {
        margin: ${theme.margin.px10} 0;
    }
    margin-right: 1rem;
`;

function Utbetalinger(props: Props) {
    return (
        <ErrorBoundary boundaryName="Utbetalinger Ytelser">
            <StyledSection aria-label={'Utbetalinger ' + props.ytelsesType}>
                <KommendeUtbetalinger kommendeUtbetalinger={props.kommendeUtbetalinger} />
                <HistoriskeUtbetalingerContainer
                    historiskeUtbetalinger={props.historiskeUtbetalinger}
                    ytelseType={props.ytelsesType}
                />
            </StyledSection>
        </ErrorBoundary>
    );
}

export default Utbetalinger;
