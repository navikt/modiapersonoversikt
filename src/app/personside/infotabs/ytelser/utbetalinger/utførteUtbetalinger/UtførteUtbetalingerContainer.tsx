import * as React from 'react';
import { AppState } from '../../../../../../redux/reducers';
import { AsyncDispatch } from '../../../../../../redux/ThunkTypes';
import { connect } from 'react-redux';
import { isLoading, isNotStarted, RestResource } from '../../../../../../redux/restReducers/restResource';
import { UtbetalingerResponse } from '../../../../../../models/utbetalinger';
import { YtelserKeys } from '../../ytelserKeys';
import {
    filtrerOgSorterUtbetalinger,
    getKnappStatus,
    inneholderToÅrGamleUtbetalinger,
    nittiDagerTilbakeITid,
    toÅrTilbakeITid
} from './utførteUtbetalingerUtils';
import PlukkRestData from '../../pleiepenger/PlukkRestData';
import { AlignTextCenter } from '../../../../../../components/common-styled-components';
import { Undertittel } from 'nav-frontend-typografi';
import ErrorBoundary from '../../../../../../components/ErrorBoundary';
import styled from 'styled-components';
import UtførteUtbetalingerListe from './UtførteUtbetalingerListe';
import {
    hentUtførteUtbetalinger,
    reloadUtførteUtbetalinger
} from '../../../../../../redux/restReducers/ytelser/utførteUtbetalinger';

export enum KnappStatus {
    Vis,
    Skjul,
    Spinner
}

interface OwnProps {
    ytelseType: YtelserKeys;
}

interface StateProps {
    utførteUtbetalinger: RestResource<UtbetalingerResponse>;
    fødselsnummer: string;
}

interface DispatchProps {
    hentNyligeUtbetalinger: (fnr: string) => void;
    hentAlleUtbetalinger: (fnr: string) => void;
}

type Props = DispatchProps & StateProps & OwnProps;

const Padding = styled.div`
    padding: 0.5rem;
`;

class UtførteUtbetalingerContainer extends React.PureComponent<Props> {
    constructor(props: Props) {
        super(props);
        this.hentToÅrGamleUtbetalinger = this.hentToÅrGamleUtbetalinger.bind(this);
    }

    componentDidMount() {
        if (isNotStarted(this.props.utførteUtbetalinger)) {
            this.props.hentNyligeUtbetalinger(this.props.fødselsnummer);
        }
    }

    hentToÅrGamleUtbetalinger() {
        if (
            isLoading(this.props.utførteUtbetalinger) ||
            inneholderToÅrGamleUtbetalinger(this.props.utførteUtbetalinger)
        ) {
            return;
        }
        this.props.hentAlleUtbetalinger(this.props.fødselsnummer);
    }

    render() {
        return (
            <ErrorBoundary boundaryName="Utførte utbetalinger">
                <section>
                    <Padding>
                        <AlignTextCenter>
                            <Undertittel tag="h4">Utførte utbetalinger</Undertittel>
                        </AlignTextCenter>
                    </Padding>
                    <PlukkRestData restResource={this.props.utførteUtbetalinger} spinnerSize="S">
                        {data => (
                            <UtførteUtbetalingerListe
                                utbetalinger={filtrerOgSorterUtbetalinger(data.utbetalinger, this.props.ytelseType)}
                                hentToÅrGamleUtbetalinger={this.hentToÅrGamleUtbetalinger}
                                knappStatus={getKnappStatus(this.props.utførteUtbetalinger)}
                            />
                        )}
                    </PlukkRestData>
                </section>
            </ErrorBoundary>
        );
    }
}

function mapStateToProops(state: AppState): StateProps {
    return {
        utførteUtbetalinger: state.restResources.utførteUtbetalingerYtelser,
        fødselsnummer: state.gjeldendeBruker.fødselsnummer
    };
}

function mapDispatchToProps(dispatch: AsyncDispatch): DispatchProps {
    return {
        hentNyligeUtbetalinger: (fnr: string) => dispatch(hentUtførteUtbetalinger(fnr, nittiDagerTilbakeITid)),
        hentAlleUtbetalinger: (fnr: string) => dispatch(reloadUtførteUtbetalinger(fnr, toÅrTilbakeITid))
    };
}

export default connect(
    mapStateToProops,
    mapDispatchToProps
)(UtførteUtbetalingerContainer);
