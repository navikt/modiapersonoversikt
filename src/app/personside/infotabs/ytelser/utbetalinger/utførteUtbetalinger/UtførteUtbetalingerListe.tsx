import * as React from 'react';
import { Knapp } from 'nav-frontend-knapper';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { FlexCenter } from '../../../../../../components/common-styled-components';
import { Utbetaling } from '../../../../../../models/utbetalinger';
import { flatMapYtelser } from '../../../utbetalinger/utils/utbetalingerUtils';
import UtførtUtbetaling from './UtførtUtbetaling';
import { KnappStatus } from './UtførteUtbetalingerContainer';

interface Props {
    utbetalinger: Utbetaling[];
    knappStatus: KnappStatus;
    hentToÅrGamleUtbetalinger: () => void;
}

function UtførteUtbetalingerListe(props: Props) {
    if (props.knappStatus === KnappStatus.Skjul && (!props.utbetalinger || props.utbetalinger.length === 0)) {
        return <AlertStripeInfo>Kunne ikke finne noen utførte utbetalinger</AlertStripeInfo>;
    }

    const utførtUtbetaling = props.utbetalinger.map(utbetaling =>
        flatMapYtelser([utbetaling]).map((ytelse, index) => (
            <UtførtUtbetaling ytelse={ytelse} utbetaling={utbetaling} key={index} />
        ))
    );

    const hentAlleUtbetalingerKnapp = !(props.knappStatus === KnappStatus.Skjul) && (
        <FlexCenter>
            <Knapp onClick={props.hentToÅrGamleUtbetalinger} spinner={props.knappStatus === KnappStatus.Spinner}>
                Hent alle utførte utbetalinger
            </Knapp>
        </FlexCenter>
    );

    return (
        <>
            <ol>{utførtUtbetaling}</ol>
            {hentAlleUtbetalingerKnapp}
        </>
    );
}

export default UtførteUtbetalingerListe;
