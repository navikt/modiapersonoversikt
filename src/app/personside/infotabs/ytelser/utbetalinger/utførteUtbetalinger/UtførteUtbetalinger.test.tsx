import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { statiskMockUtbetaling, statiskMockYtelse } from '../../../../../../mock/statiskMockUtbetaling';
import { Utbetaling } from '../../../../../../models/utbetalinger';
import { YtelserKeys } from '../../ytelserKeys';
import UtførteUtbetalingerListe from './UtførteUtbetalingerListe';
import { KnappStatus } from './UtførteUtbetalingerContainer';

test('Kommende utbetalinger matcher snapshot', () => {
    const utbetaling: Utbetaling = {
        ...statiskMockUtbetaling,
        ytelser: [
            {
                ...statiskMockYtelse,
                type: YtelserKeys.Foreldrepenger
            }
        ]
    };

    const result = renderer.create(
        <UtførteUtbetalingerListe
            utbetalinger={[utbetaling]}
            knappStatus={KnappStatus.Vis}
            hentToÅrGamleUtbetalinger={() => null}
        />
    );

    expect(result.toJSON()).toMatchSnapshot();
});
