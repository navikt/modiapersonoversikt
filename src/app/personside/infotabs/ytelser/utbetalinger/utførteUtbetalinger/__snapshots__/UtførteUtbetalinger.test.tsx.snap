// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kommende utbetalinger matcher snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 > li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 > li > *:last-child {
  margin-left: 1rem;
}

.c2 > li:not(:last-child) {
  border-bottom: solid 0.0625rem #c6c2bf;
  padding-bottom: 0.1rem;
  margin-bottom: 0.1rem;
}

.c1 {
  overflow: auto;
  box-shadow: 0 0 0 0.07rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c1 table {
  text-align: left;
  width: 100%;
}

.c1 table thead {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: bold;
}

.c1 table th,
.c1 table td {
  padding: 0.7rem;
  vertical-align: middle;
}

.c1 table th {
  padding-bottom: 0.7rem;
}

.c1 table tr:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

.c0 {
  white-space: nowrap;
}

.c0 > *:not(:last-child) {
  margin-bottom: 1rem;
}

<div
  className="c0"
>
  <div
    className="typo-normal c1"
  >
    <table
      role="table"
    >
      <thead
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            role="columnheader"
          >
            Type
          </th>
          <th
            role="columnheader"
          >
            Netto
          </th>
          <th
            role="columnheader"
          >
            Utb.dato
          </th>
          <th
            role="columnheader"
          >
            Periode
          </th>
          <th
            role="columnheader"
          >
            Bruttobeløp
          </th>
          <th
            role="columnheader"
          >
            Arb.giver
          </th>
          <th
            role="columnheader"
          >
            Org.nummer
          </th>
          <th
            role="columnheader"
          >
            Skattetrekk
          </th>
          <th
            role="columnheader"
          >
            Kreditortrekk
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Foreldrepenger
          </td>
          <td
            role="cell"
          >
            11,386.00 NOK
          </td>
          <td
            role="cell"
          >
            29.03.2017
          </td>
          <td
            role="cell"
          >
            09.11.2016 - 31.10.2017
          </td>
          <td
            role="cell"
          >
            13,370.00 NOK
          </td>
          <td
            role="cell"
          >
            Ensjø Bistro
          </td>
          <td
            role="cell"
          >
            1234556678235
          </td>
          <td
            role="cell"
          >
            -730.00 NOK
          </td>
          <td
            role="cell"
          >
            <ul
              className="c2"
            >
              <li>
                <span>
                  Bjerke, Berntsen and Nilsen
                </span>
                 
                <span>
                  -1,042.00
                   NOK
                </span>
              </li>
              <li>
                <span>
                  Olsen AS
                </span>
                 
                <span>
                  -212.00
                   NOK
                </span>
              </li>
            </ul>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="alertstripe alertstripe--info"
  >
    <span
      aria-label="info"
      className="alertstripe__ikon"
    >
      <svg
        focusable="false"
        height="1.5em"
        kind="info-sirkel-fyll"
        viewBox="0 0 24 24"
        width="1.5em"
      >
        <g
          fill="none"
          fillRule="nonzero"
        >
          <path
            d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
            fill="#669DB4"
          />
          <g
            fill="#FFF"
          >
            <path
              d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zM14.976 17.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007H14.976z"
            />
          </g>
        </g>
      </svg>
    </span>
    <div
      className="typo-normal alertstripe__tekst"
    >
      Viser utførte utbetalinger 90 dager tilbake i tid
    </div>
  </div>
  <div
    className="c3"
  >
    <button
      className="knapp knapp--standard"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Hent 2 år med utførte utbetalinger
    </button>
  </div>
</div>
`;
